src/org/mozilla/javascript/JavaAdapter.java

```diff
@@ -500,7 +500,10 @@ private static ObjToIntMap getObjectFunctionNames(Scriptable obj)
     private static void appendOverridableMethods(Class<?> c,
             ArrayList<Method> list, HashSet<String> skip)
     {
-        Method[] methods = c.getDeclaredMethods();
+        Method[] methods = null;
+        if(c.isInterface())methods=c.getMethods();
+        else methods=c.getDeclaredMethods();
```

Comment:
Sadly we are not at the point in this 25-year-old project to run an automatic formatting tool on new PRs, but...

Most of the other Rhino code puts spaces around equals and brackets around all if..then..else clauses. It looks like other code in this file does the same, so could you please update to either:

    if (c.isInterface()) {
      methods = c.getMethods();
    } else {
      methods = c.getDeclaredMethods();
    }

or...

`Method[] methods = c.isInterface() ? c.getMethods() : c.getDeclaredMethods;`